staging/src/k8s.io/apiserver/pkg/registry/generic/rest/proxy.go

```diff
@@ -79,7 +79,11 @@ func NewUpgradeAwareProxyHandler(location *url.URL, transport http.RoundTripper,
 // ServeHTTP handles the proxy request
 func (h *UpgradeAwareProxyHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	if len(h.Location.Scheme) == 0 {
-		h.Location.Scheme = "http"
+		if req.URL.Scheme == "" {
+			h.Location.Scheme = "http"
+		} else {
+			h.Location.Scheme = req.URL.Scheme
```

Comment:
this doesn't seem right... the scheme of the request to this server has no bearing on the supported scheme for the backend